qutip/core/data/dispatch.pyx

```diff
@@ -564,20 +369,25 @@ cdef class Dispatcher:
         return "<dispatcher: " + self.__text_signature__ + ">"
 
     def __call__(self, *args, dtype=None, **kwargs):
-        cdef list args_, dispatch
-        cdef dict kwargs_
+        cdef list dispatch = []
```

Comment:
```suggestion
        cdef list dispatch = [None] * self._n_dispatch
```
Might also be a bit faster here than append, but not sure. It doesn't change the number of objects created. If we avoid append further up, it might be consistent to avoid it here too.

In both cases, it would be awesome to avoid creating intermediate Python objects entirely but I'm not sure if `from cpython cimport PyTuple_New, PyTuple_SetItem` is necessarily a route we want to go down unless it gives a useful performance improvement.

Reply:
I tried and timed it and other variant (`boundcheck=False`, ...) and nothing had any observable effect.
I think `PyTuple_New` will create python object so it I don't expect it to give much improvement.
`dispatch`'s length is variable, so it's cleaner with `append`.

Reply:
PyTuple_New avoids the intermediate list, but I don't like it for other reasons (it does weird things with references and we don't really want to get into the job of keep track of reference counts ourselves). If you didn't see a performance benefit, happy to keep things as they are.